### PR TITLE
Add recursivity on sos upload

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
List recursively folder and subfolder:
```
$ go run main.go sos upload -r bucket-test ./github.com/exoscale/cli/cmd
0s 100 % [==============================================================] affinitygroup.go
0s 100 % [==============================================================] affinitygroup_create.go
0s 100 % [==============================================================] affinitygroup_delete.go
0s 100 % [==============================================================] affinitygroup_list.go
0s 100 % [==============================================================] api.go
0s 100 % [==============================================================] api_layout.go
0s 100 % [==============================================================] config.go
0s 100 % [==============================================================] config_delete.go
0s 100 % [==============================================================] config_list.go
0s 100 % [==============================================================] config_set.go
0s 100 % [==============================================================] config_show.go
0s 100 % [==============================================================] dns.go
0s 100 % [==============================================================] dns_add.go
0s 100 % [==============================================================] dns_create.go
0s 100 % [==============================================================] dns_delete.go
0s 100 % [==============================================================] dns_list.go
0s 100 % [==============================================================] dns_records_add.go
0s 100 % [==============================================================] dns_records_update.go
0s 100 % [==============================================================] dns_remove.go
0s 100 % [==============================================================] dns_show.go
0s 100 % [==============================================================] dns_update.go
0s 100 % [==============================================================] eip.go
0s 100 % [==============================================================] eip_associate.go
0s 100 % [==============================================================] eip_create.go
0s 100 % [==============================================================] eip_delete.go
0s 100 % [==============================================================] eip_dissociate.go
0s 100 % [==============================================================] eip_list.go
0s 100 % [==============================================================] eip_show.go
0s 100 % [==============================================================] firewall.go
0s 100 % [==============================================================] firewall_add.go
0s 100 % [==============================================================] firewall_create.go
0s 100 % [==============================================================] firewall_delete.go
0s 100 % [==============================================================] firewall_list.go
0s 100 % [==============================================================] firewall_remove.go
0s 100 % [==============================================================] firewall_show.go
0s 100 % [==============================================================] flags.go
0s 100 % [==============================================================] icmp_code.go
0s 100 % [==============================================================] icmp_type.go
0s 100 % [==============================================================] icmpcode_string.go
0s 100 % [==============================================================] icmptype_string.go
0s 100 % [==============================================================] port.go
0s 100 % [==============================================================] port_string.go
0s 100 % [==============================================================] privnet.go
0s 100 % [==============================================================] privnet_associate.go
0s 100 % [==============================================================] privnet_create.go
0s 100 % [==============================================================] privnet_delete.go
0s 100 % [==============================================================] privnet_dissociate.go
0s 100 % [==============================================================] privnet_list.go
0s 100 % [==============================================================] privnet_show.go
0s 100 % [==============================================================] privnet_test.go
0s 100 % [==============================================================] privnet_update.go
0s 100 % [==============================================================] request.go
0s 100 % [==============================================================] root.go
0s 100 % [==============================================================] serviceoffering.go
0s 100 % [==============================================================] snapshot.go
0s 100 % [==============================================================] snapshot_create.go
0s 100 % [==============================================================] snapshot_delete.go
0s 100 % [==============================================================] snapshot_list.go
0s 100 % [==============================================================] snapshot_revert.go
0s 100 % [==============================================================] sos.go
0s 100 % [==============================================================] sos_acl.go
0s 100 % [==============================================================] sos_create.go
0s 100 % [==============================================================] sos_delete.go
0s 100 % [==============================================================] sos_download.go
0s 100 % [==============================================================] sos_headers.go
0s 100 % [==============================================================] sos_list.go
0s 100 % [==============================================================] sos_metadata.go
0s 100 % [==============================================================] sos_remove.go
0s 100 % [==============================================================] sos_upload.go
0s 100 % [==============================================================] ssh.go
0s 100 % [==============================================================] sshkey.go
0s 100 % [==============================================================] sshkey_create.go
0s 100 % [==============================================================] sshkey_delete.go
0s 100 % [==============================================================] sshkey_list.go
0s 100 % [==============================================================] sshkey_upload.go
0s 100 % [==============================================================] status.go
0s 100 % [==============================================================] template.go
0s 100 % [==============================================================] template_list.go
0s 100 % [==============================================================] template_show.go
0s 100 % [==============================================================] version.go
0s 100 % [==============================================================] vm.go
0s 100 % [==============================================================] vm_create.go
0s 100 % [==============================================================] vm_delete.go
0s 100 % [==============================================================] vm_list.go
0s 100 % [==============================================================] vm_reboot.go
0s 100 % [==============================================================] vm_reset.go
0s 100 % [==============================================================] vm_resize.go
0s 100 % [==============================================================] vm_scale.go
0s 100 % [==============================================================] vm_show.go
0s 100 % [==============================================================] vm_start.go
0s 100 % [==============================================================] vm_stop.go
0s 100 % [==============================================================] vm_update_ip.go
0s 100 % [==============================================================] zone.go
Successfully uploaded "cmd"
```
Fix https://github.com/exoscale/cli/issues/59

### CODE REVIEW

**Comment by pierre-emmanuelJ** [comment]
image
Yes I have to make it Async

**Comment by pierre-emmanuelJ** [comment]
image
Thanks for this issue! I will Fix it

File: cmd/sos_upload.go [hunk 1]
```diff
@@ -125,7 +129,7 @@ var sosUploadCmd = &cobra.Command{
 		)
 		taskWG.Add(lenFileToUpload)
 
-		workerSem := make(chan int, parallelTask)
+		workerSem := make(chan int, 4)
```
Comment (greut) [inline cmd/sos_upload.go]: having a proper variable for this value helps.

Comment (pierre-emmanuelJ) [reply cmd/sos_upload.go]: yes `4` was for testing

**Pull request merged**